packages/bff/src/services/catalogService.ts

```diff
@@ -148,6 +148,53 @@ const getAttributeIds = (
   ),
 ];
 
+// Fetched all eservic consumers in a recursive way
+export const fetchAllEserviceConsumers = async (
+  catalogProcessClient: CatalogProcessClient,
+  headers: Headers,
+  eServiceId: EServiceId,
+  start: number = 0
+): Promise<catalogApi.EServiceConsumer[]> => {
+  const consumers = await getEserviceFrom(
+    catalogProcessClient,
+    eServiceId,
+    0,
+    headers
+  );
+
+  if (consumers.length >= 50) {
+    return consumers.concat(
+      await getEserviceFrom(
+        catalogProcessClient,
+        eServiceId,
+        start + 50,
+        headers
+      )
+    );
+  }
+
+  return consumers;
```

Comment:
I am not sure about this implementation.
Doubts:
- `consumers.length` will not be higher than 50. Maybe `totalCount` is the field to check from the returned value of `getEServiceConsumers`.
- if the next iteration has offset `start + 50`, we should increase `start` at each iteration

Reply:
You are right the correct way is to check `totalcount` also it's needed to invoke `getEserviceFrom` with properly offset.

Reply:
This [commit](https://github.com/pagopa/interop-be-monorepo/pull/787/commits/6494c92444dd3118ee5f32a8bbc4d996d71e8577) fix this check.

Reply:
Thanks, but I am still not sure, sorry.
Do you mean to use the function iteratively or recursively?
The comment on top says "in a recursive way", but the function is not calling itself.
If you want the function to be iterative, then there should be a loop, otherwise it will only perform two calls to `getEserviceFrom`. By the way, what's the meaning of the name `getEServiceFrom`?